CBinding/Gui/GeneralOptionsPanel.cs

```diff
@@ -42,93 +42,62 @@ namespace CBinding
 {
 	public partial class GeneralOptionsPanel : Gtk.Bin
 	{
-		ICompiler default_c_compiler;
-		List<ICompiler> c_compilers = new List<ICompiler> ();
-		
-		ICompiler default_cpp_compiler;
-		List<ICompiler> cpp_compilers = new List<ICompiler> ();
-		
+
+		CMakeToolchain default_toolchain;
+		List<CMakeToolchain> toolchain = new List<CMakeToolchain> ();
+
 		public GeneralOptionsPanel ()
 		{
 			this.Build ();
-			
-			object[] compilers = AddinManager.GetExtensionObjects ("/CBinding/Compilers");
-		
-			foreach (ICompiler compiler in compilers) {
-				if (compiler.Language == Language.C) {
-					c_compilers.Add (compiler);
-				} else if (compiler.Language == Language.CPP) {
-					cpp_compilers.Add (compiler);
+
+			object [] toolchains = AddinManager.GetExtensionObjects ("/CBinding/Toolchains");
+			foreach (CMakeToolchain Toolchain in toolchains) {
+				toolchain.Add (Toolchain); 
+			}
+
+			foreach (CMakeToolchain Toolchain in toolchain) {
+				if (Toolchain.IsSupported) {
+					cCombo.AppendText (Toolchain.ToolchainName);
 				}
 			}
+				
 			
-			foreach (ICompiler compiler in c_compilers)
-				cCombo.AppendText (compiler.Name);
-			
-			foreach (ICompiler compiler in cpp_compilers)
-				cppCombo.AppendText (compiler.Name);
-			
-			string c_compiler = PropertyService.Get<string> ("CBinding.DefaultCCompiler", new GccCompiler ().Name);
-			string cpp_compiler = PropertyService.Get<string> ("CBinding.DefaultCppCompiler", new GppCompiler ().Name);
+			string TChain = PropertyService.Get<string> ("CBinding.DefaultToolchain", new MinGW32Toolchain ().ToolchainName);
```

Comment:
MinGW32Toolchain is Windows-specific. The default toolchain should be the one returned by GetDefaultToolchain().

Probably something like this?

```
string toolchainName = PropertyService.Get<string> ("CBinding.DefaultToolchain", null)
    ?? GetDefaultToolchain().ToolchainName;
```